### PR TITLE
Trace propagation support

### PR DESCRIPTION
**Pull request by viveklak**

### CODE REVIEW

File: provider/pkg/googleclient/http.go [hunk 1]
```diff
@@ -122,7 +123,9 @@ func (c *GoogleClient) RequestWithTimeout(
 		return nil, err
 	}
 	// TODO: request not handling timeout
-	req, err := http.NewRequest(method, u, &buf)
+	// ctx, cancelFunc := context.WithTimeout(ctx, timeout)
+	// defer cancelFunc()
```
Comment (lblackstone) [inline provider/pkg/googleclient/http.go:127]: Still needed?

File: provider/pkg/googleclient/trace.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		return roundTripFunc(func(req *http.Request) (*http.Response, error) {
+			ctx := req.Context()
+			if ctx == nil {
+				return rt.RoundTrip(req)
+			}
+
+			name := fmt.Sprintf("%s@%s", req.Method, req.URL.Path)
+			span, ctx := opentracing.StartSpanFromContext(ctx, name)
+			defer span.Finish()
+			req2 := req.WithContext(ctx)
```
Comment (lblackstone) [inline provider/pkg/googleclient/trace.go:31]: I'd suggest a more descriptive name for `req2`. Maybe `reqWithContext`?

**Review by lblackstone — APPROVED** [review]
Couple of small comments, but LGTM overall

**Comment by mjeffryes** [comment]
closing as stale

**Pull request closed without merging**